jcl/src/java.base/share/classes/java/lang/Class.java

```diff
@@ -1454,19 +1454,21 @@ private Method throwExceptionOrReturnNull(boolean throwException, String name, C
 Method getMethodHelper(
 	boolean throwException, boolean forDeclaredMethod, List<Method> methodList, String name, Class<?>... parameterTypes)
 	throws NoSuchMethodException {
-	Method result;
-	Method bestCandidate;
+	Method result = null;
+	Method resultFromSuperClass = null;
+	Method bestCandidate = null;
+	Method candidateFromSuperClass = null;
 	String strSig;
 	
 	/*[PR CMVC 114820, CMVC 115873, CMVC 116166] add reflection cache */
-	if (parameterTypes == null) {
+	if (null == parameterTypes) {
```

Comment:
There is no to put r-value in the left for **Java** equal comparison. Same for following similar changes.

Reply:
Agreed and restored them back.